src/hip_memory.cpp

```diff
@@ -690,29 +694,28 @@ hipError_t hipMallocArray(hipArray** array, const hipChannelFormatDesc* desc, si
     return ihipLogStatus(hip_status);
 }
 
-hipError_t hipArray3DCreate(hipArray** array, const HIP_ARRAY_DESCRIPTOR* pAllocateArray) {
+hipError_t hipArray3DCreate(hipArray** array, const HIP_ARRAY3D_DESCRIPTOR* pAllocateArray) {
     HIP_INIT_SPECIAL_API(hipArray3DCreate, (TRACE_MEM), array, pAllocateArray);
     hipError_t hip_status = hipSuccess;
 
     auto ctx = ihipGetTlsDefaultCtx();
 
     *array = (hipArray*)malloc(sizeof(hipArray));
-    array[0]->type = pAllocateArray->flags;
-    array[0]->width = pAllocateArray->width;
-    array[0]->height = pAllocateArray->height;
-    array[0]->depth = pAllocateArray->depth;
-    array[0]->drvDesc = *pAllocateArray;
+    array[0]->type = pAllocateArray->Flags;
+    array[0]->width = pAllocateArray->Width;
+    array[0]->height = pAllocateArray->Height;
+    array[0]->depth = pAllocateArray->Depth;
```

Comment:
Below properties are not set and being accessed in ihipBindTextureToArrayImpl. So when 3D array is mapped to textures it fails with segmentation fault.
   array[0]->Format = pAllocateArray->Format;
    array[0]->NumChannels = pAllocateArray->NumChannels;

Reply:
Thanks, fixed.